src/client/app/components/UIOptionsComponent.jsx

```diff
@@ -4,11 +4,42 @@
 
 import React from 'react';
 import Slider from 'react-rangeslider';
+import { MultiSelect } from 'react-selectize';
+import 'react-selectize/themes/bootstrap3.css';
 import moment from 'moment';
 import 'react-rangeslider/lib/index.css';
 import { chartTypes } from '../reducers/graph';
 import ExportContainer from '../containers/ExportContainer';
 
+// Signifies that the containing object represents a meter
+const DATA_TYPE_METER = 'DATA_TYPE_METER';
+// Signifies that the containing object represents a group of meters and groups
+const DATA_TYPE_GROUP = 'DATA_TYPE_GROUP';
+
+/**
+ * Put item's id field in tgt if the item specifies a meter
+ * @param {int[]} tgt The array to perhaps insert an item into
+ * @param {{String, int}} item The item being considered
+ * @return {Array} The modified tgt array
+ */
+function metersFilterReduce(tgt, item) {
```

Comment:
Don't put functions like this outside the component. If you need a function, put it as a method in the component (and potentially bind it in the constructor like `this.myNewFunction = this.myNewFunction.bind(this);`

However, these two functions are only used once. Consider just inlining the code in `handleDatasourceSelect()`

Reply:
These functions aren't in any way related to the component; why would they need to be associated with it? That's not OOP at all.